### PR TITLE
[core] Add GPU copy support for RExt

### PR DESCRIPTION
**Pull request by mvafin**

### CODE REVIEW

File: _studio/shared/include/cm_mem_copy.h [hunk 1]
```diff
@@ -67,6 +67,11 @@ typedef mfxI32 cmStatus;
 #define min(a,b)            (((a) < (b)) ? (a) : (b))
 #endif
 
+static bool operator < (const mfxHDLPair & l, const mfxHDLPair & r)
```
Comment (dmitryermilov) [inline _studio/shared/include/cm_mem_copy.h]: what is the idea of < operator ? How is it used?

File: _studio/shared/src/cm_mem_copy.cpp [hunk 2]
```diff
@@ -2125,7 +2578,7 @@ mfxStatus CmCopyWrapper::Initialize(eMFXHWType hwtype)
     if (m_HWType == MFX_HW_UNKNOWN)
         return MFX_ERR_UNDEFINED_BEHAVIOR;
 
-    m_timeout = CM_MAX_TIMEOUT_MS;
+    m_timeout = m_HWType >= MFX_HW_ICL ? CM_MAX_TIMEOUT_SIM : CM_MAX_TIMEOUT_MS;
```
Comment (dmitryermilov) [inline _studio/shared/src/cm_mem_copy.cpp]: why? please add a comment in code.

File: _studio/shared/src/cm_mem_copy.cpp [hunk 3]
```diff
@@ -2162,6 +2615,14 @@ mfxStatus CmCopyWrapper::InitializeSwapKernels(eMFXHWType hwtype)
     case MFX_HW_CFL:
         cmSts = m_pCmDevice->LoadProgram((void*)skl_copy_kernel_genx,sizeof(skl_copy_kernel_genx),m_pCmProgram,"nojitter");
         break;
+#if !defined(ANDROID)
```
Comment (dmitryermilov) [inline _studio/shared/src/cm_mem_copy.cpp]: why is it under #if !defined(ANDROID)?

Comment (mvafin) [reply _studio/shared/src/cm_mem_copy.cpp]: Removed.

File: _studio/shared/src/cm_mem_copy.cpp (hunk 2 above)
Comment (mvafin) [reply _studio/shared/src/cm_mem_copy.cpp]: Removed. It was needed for simulation.

File: _studio/shared/include/cm_mem_copy.h (hunk 1 above)
Comment (mvafin) [reply _studio/shared/include/cm_mem_copy.h]: It is used to add pair to std::map in CmCopyWrapper::CreateCmSurface2D

Comment (dmitryermilov) [reply _studio/shared/include/cm_mem_copy.h]: Still not clear to me. Oleg, do you see the point?

Comment (onabiull) [reply _studio/shared/include/cm_mem_copy.h]: std::map keys should be comparable using operator <. Still I'd request to add explicit comment in the code to reduce confusion.

Comment (dmitryermilov) [reply _studio/shared/include/cm_mem_copy.h]: I see now.  Thanks. Yes, Maxim, please add a comment.

Comment (mvafin) [reply _studio/shared/include/cm_mem_copy.h]: Done.

**Comment by onabiull** [comment]
I believe PR #490 already contains this change.

**Pull request closed without merging**